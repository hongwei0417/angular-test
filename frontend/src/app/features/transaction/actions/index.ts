import * as TxnApiActions from './txn-api.actions';
import * as TxnFormPageActions from './txn-form-page.actions';
import * as TxnListPageActions from './txn-list-page.actions';
import * as TxnSettingFormActions from './txn-setting-form.actions';

export {
  TxnApiActions,
  TxnFormPageActions,
  TxnListPageActions,
  TxnSettingFormActions,
};
