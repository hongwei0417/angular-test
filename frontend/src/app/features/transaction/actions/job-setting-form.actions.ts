import { createAction, props } from '@ngrx/store';

export const addJobSetting = createAction(
  '[JobSetting] add a job setting option'
);

export const removeJobSetting = createAction(
  '[JobSetting] remove a job setting option',
  props<{ id: number }>()
);
