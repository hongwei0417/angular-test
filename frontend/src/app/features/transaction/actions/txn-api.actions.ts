import { ITransaction } from '../models/Transaction';
import { createAction, props, union } from '@ngrx/store';

export const loadTxnApiSuccess = createAction(
  '[TxnApi] Load Transaction Success',
  props<{ txns: ITransaction[] }>()
);

export const loadTxnApiFailure = createAction(
  '[TxnApi] Load Transaction Failure',
  props<{ error: any }>()
);

export const deleteTxnApiSuccess = createAction(
  '[TxnApi] Delete Transaction Failure',
  props<{ id: string }>()
);

export const createTxnApiSuccess = createAction(
  '[TxnApi] Create Transaction Success',
  props<{ txn: ITransaction }>()
);

export const createTxnApiFail = createAction(
  '[TxnApi] Create Transaction Failure',
  props<{ error: any }>()
);

const all = union({
  createTxnApiSuccess,
  createTxnApiFail,
  loadTxnApiSuccess,
  loadTxnApiFailure,
  deleteTxnApiSuccess,
});
export type TxnApiActionUnion = typeof all;
