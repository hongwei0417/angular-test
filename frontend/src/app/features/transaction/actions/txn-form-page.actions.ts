import { createAction, props, union } from '@ngrx/store';

export const createTxn = createAction(
  '[TxnFormPage] Create Transaction',
  props<{ title: string; content: string; executeCount: number }>()
);

export const clearTxnCreatePageState = createAction(
  '[TxnFormPage] Clear TxnCreate Page State'
);
