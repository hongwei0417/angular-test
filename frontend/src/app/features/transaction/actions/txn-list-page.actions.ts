import { createAction, props } from '@ngrx/store';

export const loadTxnList = createAction(
  '[TxnListPage] Load txn list table data'
);

export const clearTxnListPageState = createAction(
  '[TxnListPage] Clear txn list page state'
);

export const createNewTxn = createAction(
  '[TxnListPage] Create specific transaction info'
);

export const viewTxnInfo = createAction(
  '[TxnListPage] View specific transaction info',
  props<{ id: string }>()
);

export const editTxnInfo = createAction(
  '[TxnListPage] Edit specific transaction info',
  props<{ id: string }>()
);

export const copyTxnInfo = createAction(
  '[TxnListPage] Copy specific transaction info',
  props<{ id: string }>()
);

export const executeTxn = createAction(
  '[TxnListPage] Run specific transaction',
  props<{ id: string }>()
);

export const reloadTxn = createAction(
  '[TxnListPage] Reload specific transaction',
  props<{ id: string }>()
);
