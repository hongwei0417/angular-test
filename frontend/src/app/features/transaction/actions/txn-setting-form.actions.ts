import { createAction, props } from '@ngrx/store';

export const addFrequencySetting = createAction(
  '[TxnSetting] Add a frequency setting'
  // props<{ name: string }>()
);
