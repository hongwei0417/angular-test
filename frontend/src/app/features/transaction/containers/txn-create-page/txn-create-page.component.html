<!-- <h3 class="text-center">Txn Create Page</h3>

<div class="d-flex justify-content-center">
  <button [routerLink]="['/transaction','list']">Transaction list</button>
  <button [routerLink]="['/transaction','create']">Create Transaction</button>
</div> -->

<app-txn-create
  [errorMessage]="(error$ | async)!"
  (submitted)="onSubmit($event)"
  (clearState)="onClearState()"
></app-txn-create>
