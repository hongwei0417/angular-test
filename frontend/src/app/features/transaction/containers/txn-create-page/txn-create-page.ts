import { Observable, of } from 'rxjs';
import { Store, select } from '@ngrx/store';
import { Component, OnDestroy, OnInit } from '@angular/core';
import { TxnCreatePageActions } from '../../actions';
import * as fromTxn from '../../reducers';

@Component({
  selector: 'app-txn-create-page',
  templateUrl: './txn-create-page.component.html',
  styleUrls: ['./txn-create-page.component.scss'],
})
export class TxnCreatePageComponent implements OnInit, OnDestroy {
  error$!: Observable<string>;

  constructor(private store$: Store<fromTxn.State>) {}

  ngOnInit(): void {
    this.error$ = this.store$.pipe(select(fromTxn.getTxnCreateError));
  }

  ngOnDestroy(): void {
    this.store$.dispatch(TxnCreatePageActions.clearTxnCreatePageState());
  }

  onSubmit({ title, content, executeCount }: any): void {
    this.store$.dispatch(
      TxnCreatePageActions.createTxn({ title, content, executeCount })
    );
  }
}
