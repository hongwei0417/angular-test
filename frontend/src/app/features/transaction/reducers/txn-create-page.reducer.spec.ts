import { TxnApiActions, TxnCreatePageActions } from '../actions';
import { reducer, initialState } from './txn-create-page.reducer';

describe('Txn Create Page Reducer', () => {
  describe('unknown action', () => {
    it('should return the previous state', () => {
      const action = {} as any;

      const result = reducer(initialState, action);

      expect(result).toBe(initialState);
    });
  });

  describe('create txn', () => {
    it('should update loading status while creating', () => {
      const action = TxnCreatePageActions.createTxn({
        title: 'Jest Testing',
        content: 'This is a test case',
        executeCount: 10
      });
      const result = reducer(initialState, action);
      expect(result.loading).toBe(true);
    });
  });
});
