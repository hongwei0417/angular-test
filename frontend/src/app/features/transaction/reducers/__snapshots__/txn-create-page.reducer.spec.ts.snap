// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Txn Create Page Reducer unknown action should return the previous state 1`] = `
Object {
  "error": "this is a error",
  "loading": true,
}
`;
